.github/scripts/make_mocs.py

```diff
@@ -0,0 +1,233 @@
+        return output
+
+    def make_moc_for_directory(self, root, dirs, files):
+        result = ''
+
+        sorted_dirs = sorted(dirs, key=str.casefold)
+        result += self.make_moc_for_sub_directories(root, sorted_dirs)
+
+        sorted_files = sorted(files, key=str.casefold)
+        result += self.make_moc_for_files(root, sorted_files)
+
+        if not result:
+            result = '\n'
+        return result
+
+    def make_moc_for_directory_with_delimiters(self, root, dirs, files):
+        result = ''
+        result += MocDelimiter().initial_delimiter()
+        result += self.make_moc_for_directory(root, dirs, files)
+        result += MocDelimiter().final_delimiter()
+        return result
+
+    def update_existing_moc(self, initial_content, new_moc_content_with_delimiters):
+        inside_old_index = False
+        index_written = False
+        result = ''
+        for line in initial_content:
+
+            if line == MocDelimiter().initial_delimiter():
+                inside_old_index = True
+                result += new_moc_content_with_delimiters
+                index_written = True
+                continue
+
+            if line == MocDelimiter().final_delimiter():
+                inside_old_index = False
+                continue
+
+            if not inside_old_index:
+                result += line
+        if not index_written:
+            result += new_moc_content_with_delimiters
+
+        return result
+
+    def make_line_for_file(self, directory, file):
+        link_name, extension = os.path.splitext(file)
+        if extension != '.md':
+            link_name += extension
+        return self.make_link_line(directory, link_name)
+
+    def make_line_for_sub_directory(self, directory, sub_directory):
+        path = directory + '/' + sub_directory
+        namer = MocFileNamer()
+        file = namer.moc_name_for_sub_directory(sub_directory)
+        return self.make_link_line(path, file)
+
+    def strip_parent_directories_from_directory(self, directory):
+        # Ugly hack because all directory names start with '../../'
+        result = directory.replace('../', '')
+        if result == '..':  # Even more horrible hack for files in root of repo
+            result = ''
+        return result
+
+    def make_link_line(self, directory, link_name):
+        adjusted_directory = self.strip_parent_directories_from_directory(directory)
+        if len(adjusted_directory) > 0:
+            adjusted_directory += '/'
+        result = f'-  [[{adjusted_directory}{link_name}|{link_name}]]\n'
+        # print(f'directory={directory}\nadjusted_directory={adjusted_directory}\nlink_name={link_name}\n=> {result}')
+        return result
+
+
+class MocFileAndDirectoryFilter:
+    """Various filtering functions, to determine what is included in the generated MOC"""
+
+    def __init__(self):
+        self.DIRECTORIES_TO_EXCLUDE = ['meta-notes', 'venv']  # Directories beginning '.' are also excluded
```

Comment:
You can remove `meta-notes` from this, this was a local folder only I had and I've removed it from the vault since.

Reply:
OK, thanks, I will remove meta-notes.

Reply:
met-notes removed.